control-plane-operator/controllers/hostedcontrolplane/kas/service.go

```diff
@@ -40,6 +40,9 @@ func ReconcileService(svc *corev1.Service, strategy *hyperv1.ServicePublishingSt
 	case hyperv1.LoadBalancer:
 		if isPublic {
 			svc.Spec.Type = corev1.ServiceTypeLoadBalancer
+			if strategy.LoadBalancer != nil && strategy.LoadBalancer.Hostname != "" {
+				svc.ObjectMeta.Annotations[hyperv1.ExternalDNSHostnameAnnotation] = strategy.LoadBalancer.Hostname
```

Comment:
Shouldn't we nilcheck `Annotations` first?

Reply:
But it just stomps it.  I'll fix it up.